src/mixins/IntrusiveDeleter.h

```diff
@@ -1,34 +1,35 @@
-// Copyright 2004-present Facebook. All Rights Reserved.
-
-#pragma once
-
-#include <cassert>
-#include <cstddef>
-#include <memory>
-
-namespace reactivesocket {
-
-/// Deletes itself using the first time the reference count, initially one,
-/// drops to zero.
-class IntrusiveDeleter {
- public:
-  virtual ~IntrusiveDeleter() = 0;
-
-  void incrementRefCount() {
-    ++refCount_;
-  }
-
-  void decrementRefCount() {
-    assert(refCount_ > 0);
-    if (--refCount_ == 0) {
-      // Tail-call
-      delete this;
-    }
-  }
-
- private:
-  size_t refCount_{1};
-};
-
-inline IntrusiveDeleter::~IntrusiveDeleter() = default;
-}
+//TODO: will delete in the next round
+//// Copyright 2004-present Facebook. All Rights Reserved.
+//
+//#pragma once
+//
+//#include <cassert>
+//#include <cstddef>
+//#include <memory>
+//
+//namespace reactivesocket {
+//
+///// Deletes itself using the first time the reference count, initially one,
+///// drops to zero.
+//class IntrusiveDeleter {
```

Comment:
The typical circular reference in the system happens when you implement a class implementing Subscriber and Subscription interfaces and have the class be a middle man between 2 subscribers.
I tested that scenario and verified that since we are releasing smart pointers on the terminating signal, we break any circular references.